stetho/src/test/java/com/facebook/stetho/inspector/network/AsyncPrettyPrintResponseBodyTest.java

```diff
@@ -0,0 +1,243 @@
+    AsyncPrettyPrinter mAsyncPrettyPrinter = mPrettyPrinterTestFactory.getInstance(
+        TEST_HEADER_NAME,
+        TEST_HEADER_VALUE);
+    assert(mPrettyPrinterTestFactory != null);
+    assert(mAsyncPrettyPrinter != null);
+    mAsyncPrettyPrinter.printTo(writer, mInputStream);
+    assertEquals(PRETTY_PRINT_PREFIX + Arrays.toString(TEST_RESPONSE_BODY), out.toString());
+  }
+
+  @Test
+  public void testInitAsyncPrettyPrinterForResponseWithRegisteredHeader() {
+    ArrayList<String> headerNames = new ArrayList<String>();
+    ArrayList<String> headerValues = new ArrayList<String>();
+
+    headerNames.add("unregistered header name 1");
+    headerNames.add("unregistered header name 2");
+    headerNames.add(TEST_HEADER_NAME);
+    headerValues.add("unregistered header value 1");
+    headerValues.add("unregistered header value 2");
+    headerValues.add(TEST_HEADER_VALUE);
+
+    TestInspectorResponse testResponse = new TestInspectorResponse(
+        headerNames,
+        headerValues,
+        TEST_REQUEST_ID
+    );
+    NetworkEventReporterImpl.initAsyncPrettyPrinterForResponse(
+        testResponse,
+        mNetworkPeerManager);
+    verify(mResponseBodyFileManager, times(1)).associateAsyncPrettyPrinterWithId(
+        eq(TEST_REQUEST_ID),
+        any(AsyncPrettyPrinter.class)
+    );
+  }
+
+  @Test
+  public void testInitAsyncPrettyPrinterForResponseWithUnregisteredHeader() {
+    ArrayList<String> headerNames = new ArrayList<String>();
+    ArrayList<String> headerValues = new ArrayList<String>();
+
+    headerNames.add("unregistered header name 1");
+    headerNames.add("unregistered header name 2");
+    headerNames.add("unregistered header name 3");
+    headerValues.add("unregistered header value 1");
+    headerValues.add("unregistered header value 2");
+    headerValues.add("unregistered header value 3");
+
+    TestInspectorResponse testResponse = new TestInspectorResponse(
+        headerNames,
+        headerValues,
+        TEST_REQUEST_ID
+    );
+    NetworkEventReporterImpl.initAsyncPrettyPrinterForResponse(
+        testResponse,
+        mNetworkPeerManager);
+    verify(mResponseBodyFileManager, never()).associateAsyncPrettyPrinterWithId(
+        eq(TEST_REQUEST_ID),
+        any(AsyncPrettyPrinter.class)
+    );
+  }
+
+  private class PrettyPrinterTestFactory extends DownloadingAsyncPrettyPrinterFactory {
+    @Override
+    protected void doPrint(PrintWriter output, InputStream payload, String schema)
+        throws IOException {
+      ByteArrayOutputStream out = new ByteArrayOutputStream();
+      Util.copy(payload, out, new byte[1024]);
+      String prettifiedContent = PRETTY_PRINT_PREFIX + Arrays.toString(out.toByteArray());
+      output.write(prettifiedContent);
+      output.close();
+    }
+
+    @Override @Nullable
+    protected MatchResult matchAndParseHeader(String headerName, String headerValue) {
+      return new MatchResult("https://www.facebook.com", PrettyPrinterDisplayType.TEXT);
```

Comment:
Your test should specify what `headerName` is before returning the `MatchResult` This way you can verify what happens if there are no matches.